### PR TITLE
Return Moffins and Mothmellows

### PR DESCRIPTION
**Pull request by Wolfking6116**
## About the PR

Frees Moth's (and others) from the Tyranny of their stomachs.

Makes several important items edible once again.

## Why / Balance

Back by popular demand! After many months of waiting.

## Technical details

~~Special bonus is for Darin.~~

## Media

Content Client_YzxmjxYR8w

Content Client_4PeEylPcu0

Content Client_16HD1fHKDk

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

- tweak: Moths are no longer Tryannized by their stomachs. Moffins and Mothmallows are starting to look particularly appetizing... (and not just to Moth's!)

### CODE REVIEW

File: Resources/Prototypes/Nyanotrasen/Entities/Objects/Specific/Mail/base_mail.yml [hunk 1]
```diff
@@ -95,7 +95,7 @@
         Blunt: 10
   - type: CargoSellBlacklist
   - type: Food # Frontier - Moth food
-    requiresSpecialDigestion: true
+  #  requiresSpecialDigestion: true
   - type: SolutionContainerManager
     solutions:
       food:
```
Comment (dvir001) [inline Resources/Prototypes/Nyanotrasen/Entities/Objects/Specific/Mail/base_mail.yml:101]: Dont mess with this one, we have a conflicted rework

**Comment by dvir001** [comment]
Also as a note, when you made this change, did it make it so EVERYONE can now eat them?
Not a good fix

The food system we working on should address this without tags conflicts.

File: Resources/Prototypes/Nyanotrasen/Entities/Objects/Specific/Mail/base_mail.yml (hunk 1 above)
Comment (Wolfking6116) [reply Resources/Prototypes/Nyanotrasen/Entities/Objects/Specific/Mail/base_mail.yml:101]: Can do, just thought you'd get a laugh out of it!

**Comment by Wolfking6116** [comment]
> Also as a note, when you made this change, did it make it so EVERYONE can now eat them? Not a good fix

This is how it was for many many months before the enforced diet was attempted on Frontier. Everyone was able to eat them then, so it stands to reason that there isn't much reason why everyone shouldn't be able to now. 

Even with the food rework _potentially_ a month a way, it is a month less of several items contributing to entity bloat during the shifts from people receiving them in the mail and dumping them on the floor because no one is able to eat them at all.

**Comment by dvir001** [comment]
> > Also as a note, when you made this change, did it make it so EVERYONE can now eat them? Not a good fix
> 
> This is how it was for many many months before the enforced diet was attempted on Frontier. Everyone was able to eat them then, so it stands to reason that there isn't much reason why everyone shouldn't be able to now.
> 
> Even with the food rework _potentially_ a month a way, it is a month less of several items contributing to entity bloat during the shifts from people receiving them in the mail and dumping them on the floor because no one is able to eat them at all.
> 
> The problem this seeks to solve is that these foods are entirely inedible at current. It is a viable fix, albeit an inelegant one. It'll be great to see them polished in the rework, but for now seeing them go completely unused is a waste and a detriment.

As I said on discord, everyone able to eat them was not intended, its taking away from moth unique food

It still not fixed as the actual fix be a return to allow moth only, we cannot fix it without a new system since it will also effect all cloth in the game base on existing tags if we wont build new code for it.

Having this issue exist for another month isn't an issue considering it wont be a change of "Can eat it now" only to again won't be able to eat it, confusing players more.

Why is this now an urgent issue when it was not urgent for months? how is not waiting a single month going to be so much of an issue that were willing to make a new issue now of "Admin why I cannot eat this now I could have eat this a day before" case.

 Also, if you want to fix the food that have no cloth in it (Like the cookies) this is fine, they can be everyone food since they dont have cloth to make them moth only.
 
 Everyone eating cloth based food isnt good for RP also.

**Comment by Wolfking6116** [comment]
> As I said on discord, everyone able to eat them was not intended, as it's taking away from moth unique food.

Being inedible is already taking away from moth unique food. You also said on discord that part of your reservation to this change is to keep things the same as upstream. These are Nyano files, not Wizden.

> This PR still not fixing the actual issue of the making the moth food, moth only, and we cannot fix this issue without a new system since it will also affect all cloth in the game base on existing tags, if we won't build some new code for it.

That's outside of my abilities at this time, and I have been told that even if it were within them it would be futile with the system You and Erhard have been working on.

> Having this issue exist for another month isn't an issue considering it wont be a change of "Can eat it now" only to again won't be able to eat it, confusing players more.

There were little to no people who were upset that they could no longer eat these when it was diet locked before, and since its 'revert' the only people who have been upset about not being able to eat them have been Moth players.
 
> Why is this now an urgent issue when it was not urgent for months? how is not waiting a single month going to be so much of an issue that were willing to make a new issue now of "Admin why I cannot eat this now I could have eat this a day before" case.
 
It isn't urgent. It has been requested for months at this point, repeatedly, several times a week, even from members of the admin team. I was unable to fix it before. https://discord.com/channels/1123826877245694004/1123851369913266256/1186093816139231284 - my first attempt at it, knowing barely anything about how things worked, was back in December.

But now, after having done a few other PR's and dug through the code thoroughly, I can understand what was necessary to restore it, and how simple of a potential fix it could be.

> Also, if you want to fix the food that have no cloth in it (Like the cookies) this is fine, they can be everyone food since they dont have cloth to make them moth only.

It's great that those can be added, but it doesn't provide a solution for the other food items aside from waiting. 

I am fully confident in Erhard's abilities to do the Food Rework justice, and with you working with him on it I know that it'll be a fantastic work that will bring a lot of depth to the server without arbitrary restrictions and unintuitive mechanics. However, you have a lot on your plate already. You've said yourself that you haven't been around as often and even then you have a lot of different projects at hand on top of being a Maintainer. Checkraze has stated there is another upstream merge coming in the next month. The talks, discussions, and possible implementations of rebasing to E.E.

The food rework is going to be fantastic when it's implemented. But I genuinely am skeptical about it only taking a month. After all, surgery is only a month away, too.
 
> Everyone eating cloth based food isnt good for RP also.

Everyone won't be, just as they weren't for the eight months or so from Frontier's birth to the attempt at enforced diets. But everyone definitely would start eating the mail, so I'll absolutely remove that.
 
> And if you still thinks that having the food right now in game is an issue, please uncomment the food and the mail that has the moffins till a real fix is done, this will be fine as a fix till we return the food as its intended.

Having the food in game isn't an issue. The food being inedible is an issue. Even if they were removed from the mail there would still be players attempting to make them via the recipes in-game and then being disappointed that they cannot eat them. To go further and comment out the recipes until a 'real fix' is done would just be a slap in the face as it truly takes the unique food.

**Comment by ErhardSteinhauer** [comment]
Closing this PR for lack of activity. Feel free to reopen it if you get back to working on it. 
P.S.: also do note that there is somewhat actively [updated PR](https://github.com/new-frontiers-14/frontier-station-14/pull/1839) that will achieve the same result.

**Pull request closed without merging**